### PR TITLE
xds extended event

### PR DESCRIPTION
**Pull request by wasaga**
## Summary

Extends XDS events to include databroker configuration version reference, along with more information regarding actual resources that are sent, event kind, and reporting instance ID. 

Note `instance_id` is not yet set part of this PR. 

## Related issues

https://github.com/pomerium/pomerium-console/issues/914
https://github.com/pomerium/pomerium-console/issues/918

## Checklist

- [x] reference any related issues
- [ ] updated docs
- [ ] updated unit tests
- [ ] updated UPGRADING.md
- [x] add appropriate tag (`improvement` / `bug` / etc)
- [x] ready for review

### CODE REVIEW

File: pkg/grpc/config/config.proto [hunk 1]
```diff
@@ -199,4 +199,20 @@ message EnvoyConfigurationEvent {
   string message = 2;
   int32 code = 3;
   repeated google.protobuf.Any details = 4;
+  // databroker config version
+  uint64 configVersion = 5;
```
Comment (calebdoxsey) [inline pkg/grpc/config/config.proto:203]: should be `config_version`

File: internal/controlplane/xdsmgr/xdsmgr.go [hunk 2]
```diff
@@ -9,16 +9,22 @@ import (
...
 	"google.golang.org/protobuf/types/known/timestamppb"
 
+	"github.com/pomerium/pomerium/internal/contextkeys"
 	"github.com/pomerium/pomerium/internal/log"
 	"github.com/pomerium/pomerium/internal/signal"
 	configpb "github.com/pomerium/pomerium/pkg/grpc/config"
 )
 
+const (
+	maxNonceCacheSize = 1 << 12
```
[the remark this answers is not in the document]
Comment (desimone) [reply internal/controlplane/xdsmgr/xdsmgr.go:25]: Wait -- do we really want the cache size to be `1 << 12`; even a single bye that would be huge.

File: internal/contextkeys/contextkeys.go [hunk 3]
```diff
@@ -0,0 +1,15 @@
+// Package contextkeys defines common context keys shared between packages
+package contextkeys
+
+type contextKey int
+
+const (
+	// DatabrokerConfigVersion identifies uint64 databroker version of the config
```
Comment (desimone) [inline internal/contextkeys/contextkeys.go:7]:
```suggestion
	// DatabrokerConfigVersion identifies the uint64 databroker version of the config
```

File: internal/controlplane/xdsmgr/xdsmgr.go [hunk 4]
```diff
@@ -35,16 +41,21 @@ type Manager struct {
...
+	nonceToConfig, _ := lru.New(maxNonceCacheSize) // the only error they return is when size is negative, which never happens
+
 	return &Manager{
 		signal:       signal.New(),
 		eventHandler: eventHandler,
 
-		nonce:     uuid.New().String(),
-		resources: resources,
+		nonceToConfig: nonceToConfig,
+		nonce:         uuid.NewString(),
```
[the remark this answers is not in the document]
Comment (wasaga) [reply internal/controlplane/xdsmgr/xdsmgr.go:57]: I have no preference. my reading of Envoy documentation is it's expected to be unique

File: internal/controlplane/xdsmgr/xdsmgr.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (wasaga) [reply internal/controlplane/xdsmgr/xdsmgr.go:25]: 4096 doesn't seem too huge, just to be safe but sure which number feels adequate?

**Review by desimone — APPROVED** [review]

**Review by calebdoxsey — APPROVED** [review]

**Pull request merged**